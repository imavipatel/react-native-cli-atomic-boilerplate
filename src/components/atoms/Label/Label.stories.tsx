import * as React from 'react';
import { storiesOf } from '@storybook/react-native';
import { Label } from './Label';
import CenterView from '../../../../storybook/utils/CenterView/CenterView';

storiesOf('Label', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('default', () => <Label>Label Text</Label>);
