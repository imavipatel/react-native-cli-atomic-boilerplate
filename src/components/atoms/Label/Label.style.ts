import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  label: {
    fontWeight: '600',
    fontSize: 16,
    color: 'black',
  },
});
