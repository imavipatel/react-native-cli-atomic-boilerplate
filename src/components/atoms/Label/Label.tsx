import React, { ReactNode } from 'react';
import { Text } from 'react-native';
import { styles } from './Label.style';

interface LabelProps {
  children: ReactNode;
}

export const Label: React.FC<LabelProps> = props => {
  return (
    <Text style={styles.label} testID="label">
      {props.children}
    </Text>
  );
};
