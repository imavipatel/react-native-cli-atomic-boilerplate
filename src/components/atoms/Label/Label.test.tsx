import React from 'react';
import { render } from '@testing-library/react-native';
import { Label } from './Label';

describe('Label', () => {
  const spyOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const tree = render(<Label>Primary Label</Label>);

    expect(tree).toMatchSnapshot();
  });
});
