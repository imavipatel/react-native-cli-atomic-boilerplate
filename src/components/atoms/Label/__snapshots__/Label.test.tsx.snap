// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render correctly 1`] = `
<Text
  style={
    Object {
      "color": "black",
      "fontSize": 16,
      "fontWeight": "600",
    }
  }
  testID="label"
>
  Primary Label
</Text>
`;
